eth/catalyst/simulated_beacon.go

```diff
@@ -192,15 +193,15 @@ func (c *SimulatedBeacon) sealBlock(withdrawals []*types.Withdrawal, timestamp u
 
 	// Mark the payload as canon
 	if isCancun {
-		txes, err := decodeTransactions(payload.Transactions)
-		if err != nil {
-			return err
-		}
 		blobHashes := make([]common.Hash, 0)
-		for _, tx := range txes {
-			blobHashes = append(blobHashes, tx.BlobHashes()...)
+		if envelope.BlobsBundle != nil {
+			hasher := sha256.New()
+			for _, commit := range envelope.BlobsBundle.Commitments {
+				c := kzg4844.Commitment(commit[:])
```

Comment:
This conversion can panic if the length of `commit` is invalid. Not sure if this is something we should care about.

Reply:
Fixed